lib/kitchen/transport/winrm.rb

```diff
@@ -87,6 +87,10 @@ def close
         def execute(command)
           return if command.nil?
           logger.debug("[WinRM] #{self} (#{command})")
+
+          if command.length > MAX_COMMAND_SIZE
+            command = run_from_file_command(command)
```

Comment:
thats a valid point. if you have alot of individual commands you are going to run separately then there is overhead creating files for each command. However compared to the overhead of going over a wire, thats extremely negligible.